### ISSUE TITLE
New update

### ISSUE THREAD

**Issue by CaucasianAsianX** [WillWilli/PlateUpEnhancementMod#19]
I believe the new update is erroring out the mod

Startup error
```
[Error  : Unity Log] InvalidProgramException: Invalid IL code in (wrapper dynamic-method) Willi.PlateUpEnhancementMod.Patches.HandleNewShopPatch:DMD<Willi.PlateUpEnhancementMod.Patches.HandleNewShopPatch::GetPostTilesDummy> (Kitchen.HandleNewShop): IL_0006: ldarg.1


Stack trace:
System.RuntimeMethodHandle.GetFunctionPointer () (at <695d1cc93cca45069c528c15c9fdd749>:0)
MonoMod.RuntimeDetour.Platforms.DetourRuntimeILPlatform.GetFunctionPointer (System.Reflection.MethodBase method, System.RuntimeMethodHandle handle) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
MonoMod.RuntimeDetour.Platforms.DetourRuntimeILPlatform.GetNativeStart (System.Reflection.MethodBase method) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
MonoMod.RuntimeDetour.DetourHelper.GetNativeStart (System.Reflection.MethodBase method) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
MonoMod.RuntimeDetour.Detour._TopApply () (at <4e2760c7517c4ea79c633d67e84b319f>:0)
MonoMod.RuntimeDetour.Detour._RefreshChain (System.Reflection.MethodBase method) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
MonoMod.RuntimeDetour.Detour.Apply () (at <4e2760c7517c4ea79c633d67e84b319f>:0)
MonoMod.RuntimeDetour.Detour..ctor (System.Reflection.MethodBase from, System.Reflection.MethodBase to, MonoMod.RuntimeDetour.DetourConfig& config) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
(wrapper dynamic-method) MonoMod.RuntimeDetour.ILHook+Context.DMD<MonoMod.RuntimeDetour.ILHook+Context::Refresh>(MonoMod.RuntimeDetour.ILHook/Context)
(wrapper dynamic-method) MonoMod.Utils.DynamicMethodDefinition.Trampoline<MonoMod.RuntimeDetour.ILHook+Context::Refresh>?976242688(object)
HarmonyLib.Internal.RuntimeFixes.StackTraceFixes.OnILChainRefresh (System.Object self) (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
MonoMod.RuntimeDetour.ILHook.Apply () (at <4e2760c7517c4ea79c633d67e84b319f>:0)
HarmonyLib.PatchFunctions.ReversePatch (HarmonyLib.HarmonyMethod standin, System.Reflection.MethodBase original, System.Reflection.MethodInfo postTranspiler, System.Reflection.MethodInfo postManipulator) (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
Rethrow as HarmonyException: IL Compile Error (unknown location)
HarmonyLib.PatchClassProcessor.ReportException (System.Exception exception, System.Reflection.MethodBase original) (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
HarmonyLib.PatchClassProcessor.Patch () (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
HarmonyLib.Harmony.<PatchAll>b__11_0 (System.Type type) (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
HarmonyLib.CollectionExtensions.Do[T] (System.Collections.Generic.IEnumerable`1[T] sequence, System.Action`1[T] action) (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
HarmonyLib.Harmony.PatchAll (System.Reflection.Assembly assembly) (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
HarmonyLib.Harmony.CreateAndPatchAll (System.Reflection.Assembly assembly, System.String harmonyInstanceId) (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
Willi.PlateUpEnhancementMod.PlateUpEnhancement.Awake () (at <b76fc010c0ea4419bac20887a1d99817>:0)
UnityEngine.GameObject:AddComponent(Type)
BepInEx.Bootstrap.Chainloader:Start()
UnityEngine.Application:.cctor()
Unity.Entities.DefaultWorldInitialization:RegisterUnloadOrPlayModeChangeShutdown()
Unity.Entities.DefaultWorldInitialization:Initialize(String, Boolean)
Unity.Entities.AutomaticWorldBootstrap:Initialize()

```

During gameplay errors
```
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
Willi.PlateUpEnhancementMod.Patches.DifficultyHelpersPatch.PatiencePlayerCountModifier_Postfix (System.Single& __result, System.Int32 player_count) (at <b76fc010c0ea4419bac20887a1d99817>:0)
(wrapper dynamic-method) Kitchen.DifficultyHelpers.DMD<Kitchen.DifficultyHelpers::PatiencePlayerCountModifier>(int)
Kitchen.UpdateCustomerImpatience.OnUpdate () (at <eaf8007f41fb4cac9a42c988797c1aeb>:0)
Unity.Entities.SystemBase.Update () (at <14b4829b90b7418182d555e6eb0e629d>:0)
Unity.Entities.ComponentSystemGroup.UpdateAllSystems () (at <14b4829b90b7418182d555e6eb0e629d>:0)
UnityEngine.Debug:LogException(Exception)
Unity.Debug:LogException(Exception)
Unity.Entities.ComponentSystemGroup:UpdateAllSystems()
Unity.Entities.ComponentSystemGroup:OnUpdate()
Unity.Entities.ComponentSystem:Update()
Unity.Entities.DummyDelegateWrapper:TriggerUpdate()

[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
Willi.PlateUpEnhancementMod.Patches.CreateCustomerSchedulePatch.DetermineTotalCustomers_Postfix (System.Single& __result, KitchenData.KitchenParameters parameters, System.Int32 player_count, System.Int32 day, KitchenData.GameDifficultySettings difficulty_settings) (at <b76fc010c0ea4419bac20887a1d99817>:0)
(wrapper dynamic-method) Kitchen.CreateCustomerSchedule.DMD<Kitchen.CreateCustomerSchedule::DetermineTotalCustomers>(KitchenData.KitchenParameters,int,int,KitchenData.GameDifficultySettings)
Kitchen.CreateCustomerSchedule.GetCustomersForDay (Kitchen.CreateCustomerSchedule system, KitchenData.KitchenParameters parameters, System.Int32 player_count, System.Int32 day, System.Action`3[T1,T2,T3] group_callback) (at <2e9254c3c81a49f7bca34755ab66c7a6>:0)
Kitchen.ParametersDisplayView+UpdateView.OnUpdate () (at <2e9254c3c81a49f7bca34755ab66c7a6>:0)
Unity.Entities.SystemBase.Update () (at <14b4829b90b7418182d555e6eb0e629d>:0)
Unity.Entities.ComponentSystemGroup.UpdateAllSystems () (at <14b4829b90b7418182d555e6eb0e629d>:0)
UnityEngine.Debug:LogException(Exception)
Unity.Debug:LogException(Exception)
Unity.Entities.ComponentSystemGroup:UpdateAllSystems()
Unity.Entities.ComponentSystemGroup:OnUpdate()
Unity.Entities.ComponentSystem:Update()
Unity.Entities.ComponentSystemGroup:UpdateAllSystems()
Unity.Entities.ComponentSystemGroup:OnUpdate()
Unity.Entities.ComponentSystem:Update()
Unity.Entities.DummyDelegateWrapper:TriggerUpdate()
```

**Comment by WillWilli** [comment]
Thanks mate. Should be fixed in latest release [v0.4.1](https://github.com/WillWilli/PlateUpEnhancementMod/releases/download/0.4.1/Willi.PlateUpEnhancementMod.0.4.1.zip)

**Comment by CaucasianAsianX** [comment]
That was fast! thanks!
